src/REL/d/a/obj/d_a_obj_distant_ship.cpp

```diff
@@ -1,3 +1,61 @@
 #include "d/a/obj/d_a_obj_distant_ship.h"
 
+#include "common.h"
+#include "d/a/obj/d_a_obj_base.h"
+#include "d/flag/sceneflag_manager.h"
+#include "f/f_base.h"
+#include "nw4r/g3d/res/g3d_resfile.h"
+#include "toBeSorted/arc_managers/current_stage_arc_manager.h"
+
 SPECIAL_ACTOR_PROFILE(OBJ_DIS_SHIP, dAcOdistantShip_c, fProfile::OBJ_DIS_SHIP, 0x143, 0, 0);
+
+bool dAcOdistantShip_c::createHeap() {
+    char *modelName = getModelName();
+    mBrres = (nw4r::g3d::ResFile)CurrentStageArcManager::GetInstance()->getData("g3d/stage.brres");
+    TRY_CREATE(mModel.create(mBrres.GetResMdl(modelName), &heap_allocator, 0x120, 1, 0));
+    return SUCCEEDED;
+}
+
+int dAcOdistantShip_c::create() {
+    // Sure, why not
+    mIsVisible = false;
+    u32 p = params;
+    u16 r = (s32)roomid;
+    mSubtype = (p >> 8) & 0xF;
+
+    bool flag = SceneflagManager::sInstance->checkFlag(r, p & 0xFF);
```

Comment:
We have a `checkBoolFlag` inline here, that might obviate the need for a separate bool

Reply:
I've changed to the `changeBoolFlag`. Messed around with various inlines and nothing seems to work nicely.

I can get rid of the `u16 r = (u32)roomid` by changing it to `u16 r = getRoomId()`. However this line needs to be called before `mSubtype` is set for the compiler to use the correct registers.

Replacing `p` with `getFromParams()` also causes incorrect registers to be used.

I've added a scratch if anyone wants to mess around with it: https://decomp.me/scratch/0qPPj